src/søknad/steg/4-barnasbosted/BarnasBosted.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useState } from 'react';
+import useSøknadContext from '../../../context/SøknadContext';
+import Side from '../../../components/side/Side';
+import { Element } from 'nav-frontend-typografi';
+import { Input } from 'nav-frontend-skjema';
+import { Checkbox } from 'nav-frontend-skjema';
+import { ISpørsmål } from '../../../models/spørsmal';
+import {
+  borINorge,
+  avtaleOmDeltBosted,
+  harAnnenForelderSamværMedBarn,
+  harDereSkriftligSamværsavtale,
+  boddSammenFør,
+  borISammeHus,
+  hvorMyeSammen
+} from './ForeldreConfig';
+import { IForelder } from '../../../models/person';
+import JaNeiSpørsmål from '../../../components/spørsmål/JaNeiSpørsmål';
+import MultiSvarSpørsmål from '../../../components/spørsmål/MultiSvarSpørsmål';
+import KomponentGruppe from '../../../components/gruppe/KomponentGruppe';
+import BarnasBostedHeader from './BarnasBostedHeader';
+import FeltGruppe from '../../../components/gruppe/FeltGruppe';
+import { useIntl } from 'react-intl';
+import Datovelger, { DatoBegrensning } from '../../../components/dato/Datovelger';
+import LocaleTekst from '../../../language/LocaleTekst';
+import { AlertStripeInfo } from 'nav-frontend-alertstriper';
+import HvordanPraktiseresSamværet from './HvordanPraktiseresSamværet';
+
+const BarnasBosted: React.FC = () => {
+  const intl = useIntl();
+  const { søknad } = useSøknadContext();
+  const [forelder, settForelder] = useState<IForelder>({});
+
+  const settHarForelderSamværMedBarn = (spørsmål: string, valgtSvar: string) => {
+    const nyForelder = {...forelder, [harAnnenForelderSamværMedBarn.spørsmål_id]: valgtSvar};
+
+    if (valgtSvar === intl.formatMessage({ id: 'barnasbosted.spm.andreForelderenSamværNei'})) {
+      delete nyForelder.harDereSkriftligSamværsavtale;
+      delete nyForelder.hvordanPraktiseresSamværet;
+    }
+
+    settForelder(nyForelder);
+  }
+
+  const settHarDereSkriftligSamværsavtale = (spørsmål: string, valgtSvar: string) => {
+    const nyForelder = {...forelder, [harDereSkriftligSamværsavtale.spørsmål_id]: valgtSvar};
+
+    if (valgtSvar !== intl.formatMessage({ id: 'barnasbosted.spm.jaIkkeKonkreteTidspunkt'})) {
+      delete nyForelder.hvordanPraktiseresSamværet;
+    }
+
+    settForelder(nyForelder);
+  }
```

Comment:
Kunne man merget disse to til én settMetode? 🤔  mm, litt usikker på o det ville vært bedre.. Men kanskje vi kunne trukket ut litt kode til egne komponenter sånn at det ikke er så altfor mye inni BarnasBosted? 😸

Reply:
Tenkte det ble mer oversiktelig å ha separate metoder, selv om det blir litt redundans.. Jeg trakk uansett ut disse metodene til en egen fil i ed76723 :)

Reply:
Endret dette i ff7bc11. Lagde heller egne komponenter, som du skrev under :)